components/scream/src/physics/shoc/shoc_functions_f90.cpp

```diff
@@ -256,5 +268,49 @@ void calc_shoc_vertflux_f(Int shcol, Int nlev, Int nlevi, Real *tkh_zi,
   pack::device_to_host({vertflux}, {shcol}, {nlevi}, inout_views, true);
 }
 
+void shoc_diag_second_moments_srf_f(Int shcol, Real* wthl, Real* uw, Real* vw, Real* ustar2, Real* wstar)
+{
+  using SHOC       = Functions<Real, DefaultDevice>;
+  using Scalar     = typename SHOC::Scalar;
+  using view_1d    = typename SHOC::view_1d<Scalar>;
+
+  view_1d wthl_d(wthl, shcol),
+          uw_d(uw, shcol),
```

Comment:
wthl, uw, vw are host pointers. So you need to copy these values into mirror views, then copy the mirror data to the device.

Reply:
If you want to be a little more efficient, you can do something like:
```
// make the mirror directly as unmanaged, rather than separate managed host view with copy from wthl
view_1d::HostMirror wthl_h(wthl, shcol);
view_1d wthl_d;
Kokkos::deep_copy(wthl_d, wthl_h);

Reply:
Please use the host_to_device and device_to_host utility functions. This is exactly what they are for.

Reply:
Since you don't want packing here, just make sure those utilities think the packsize is 1.